### ISSUE TITLE
Possibly collect/donate unknown package names from "licenses" web page

### ISSUE THREAD

**Issue by nisehime** [SteamDatabase/SteamDatabase#822]
### Feature Description

I thought about posting this on Browser Extension at first, but maybe this can be done without it as well (in token dumper for example).

SteamDB says you can donate Steam Key to retrieve package name for a package with unknown name. However, there is another source for this information - licenses page on the web (/account/licenses/), which lists all packages owned by user (I guess) and all names are displayed there. This page doesn't have associated IDs for the licenses however, which is probably the reason it was not done before and [SteamDB] link added by extension is just a search link.

I noticed that `rgOwnedPackages` in Web API returns list of package IDs in chronological order, just like licenses web page, again at least in my case. So, all that is need to be done is parse the page's package table and match it with package IDs from `rgOwnedPackages`. Profit? Needs more testing, but if your results will be the same, this can be rolled out. The only downside is that Valve can change it in the future (randomise `rgOwnedPackages`), but at least something can be collected.

**Comment by xPaw** [comment]
>  This page doesn't have associated IDs for the licenses however

That's one reason.

But the other reason is that I can't really verify the names that would be submitted (unlike tokens and keys).

I thought about it before, it's just not worth the effort.

**Comment by nisehime** [comment]
Ok, what about at least implementing this in browser extension? Instead of [SteamDB] search link having a direct subid link and package names to be displayed on SteamDB itself.

**Comment by xPaw** [comment]
Sounds like an interesting idea. Make an issue in the extension repo (and perhaps contribute with a pull request?)

Also, with this idea, since free packages have subids on the license page, the order could be verified with these.

**Comment by xPaw** [comment]
Uh. I just checked against my own account, and starting with first packages the order sadly does not match what's on the page. So this idea doesn't work unfortunately.

**Comment by nisehime** [comment]
> and perhaps contribute with a pull request

I don't have time to learn coding :(

>just checked against my own account, and starting with first packages the order sadly does not match what's on the page

Are sure? `rgOwnedPackages` is in ascending order (oldest package first) and Web page is in descending, but the order itself is not random

**Comment by xPaw** [comment]
Yes. I looked at like 10 ids, they were "roughly" in the order, but definitely not the same order as the page.

**Comment by nisehime** [comment]
This sucks then. I don't have many licenses so maybe that's why. Or I was not really careful after all. Thanks for looking into it anyway. 

I checked 50 ids from the begining and from the end of my list and everything was correct. The only problem was that I had package 0 added after I added steamos devkit to my library for some reason. And I think this still might be correct order, just the package itself is not displayed on that page
